gefserver/pier/pier.go

```diff
@@ -146,6 +150,42 @@ func (p *Pier) BuildService(buildDir string) (db.Service, error) {
 	return service, nil
 }
 
+// startTimeOutTicker starts a clock that checks if a job exceeds an execution timeout
+func (p *Pier) startTimeOutTicker(jobId db.JobID) {
+	ticker := time.NewTicker(time.Second * time.Duration(p.checkInterval))
+	go func() {
+		for range ticker.C {
+			job, err := p.db.GetJob(jobId)
+
+			if err != nil {
+				p.db.SetJobState(job.ID, db.NewJobStateError("Cannot get information about the job running", 1))
+				ticker.Stop()
+				break
+			}
+			if job.State.Code != -1 {
+				ticker.Stop()
+				break
+			}
+
+			startingTime := job.Created
+			currentTime := time.Now().UTC()
+			durationTime := time.Duration(currentTime.Sub(startingTime))
+
+			if durationTime.Seconds() >= float64(p.jobExecTimeOut) {
+				p.db.SetJobState(job.ID, db.NewJobStateError("Job execution timeout exceeded", 1))
+				ticker.Stop()
+
+				theLastContainer := job.Tasks[len(job.Tasks)-1].ContainerID
+				_, err = p.docker.client.WaitContainerOrSwarmService(string(theLastContainer), true)
+				if err != nil {
+					p.db.SetJobState(job.ID, db.NewJobStateError("Container removal failed", 1))
```

Comment:
This error would overwrite the previous error, set it as 'Job execution timeout exceeded and container removal failed'